runtime/plugin/versioncheck.lua

```diff
@@ -0,0 +1,23 @@
+if vim.g.loaded_versioncheck ~= nil then
+  return
+end
+vim.g.loaded_versioncheck = true
+
+-- detect non-interactive startups
+if vim.tbl_contains(vim.v.argv, '-l') then
+  return
+end
+
+if vim.version().prerelease ~= true then
+  return
+end
+
+local augroup = vim.api.nvim_create_augroup('versioncheck', {})
+
+vim.api.nvim_create_autocmd('CursorHold', {
```

Comment:
`CursorHold` probably isn't right for this. Why not `VimEnter`?

Reply:
Just because it was mentioned in the original issue. I think the idea is we try to avoid disrupting any user startup routine by delaying it, this is just one solution. I'll have a look at VimEnter too. Now I wonder if we can rely on UIEnter to avoid the check for non-interactive startup?